src/credential_provider.rs

```diff
@@ -58,134 +111,102 @@ impl Debug for AuthTokenSource {
 }
 
 #[derive(Debug)]
-pub struct CredentialProviderBuilder {
+struct CredentialProviderBuilder {
```

Comment:
for now I made this builder private. We use it internally, and some parts of it may become public again in the future, but I wanted to simplify the 90% case for creating a CredentialProvider for now.